src/components/Common/ConnectionBanner.tsx

```diff
@@ -1,10 +1,15 @@
 import React from "react";
 import { Button, Row, Col } from "react-bootstrap";
 
-export default function ConnectionBanner({ loading, connected, reconnect }) {
-  return loading || connected ? (
-    ""
-  ) : (
+type Props = {
+  loading: boolean;
+  connected: boolean;
+  reconnect: () => void;
+};
```

Comment:
If I see it correctly, it can't be loading and connected at the same time. 
So there's `loading` or `connected` or `disconnected` (or something like that)
If you have just one string instead of multiple booleans, there can't be any wrong states.
In Typescript you can write a type like
```Typescript
type ConnectionStatus = 'connected' | 'loading' | 'disconnected'
```